drone/src/agent/executor.rs

```diff
@@ -89,6 +86,28 @@ impl<E: Engine> Clone for Executor<E> {
     }
 }
 
+async fn update_backend_state(
+    nc: &TypedNats,
+    state: BackendState,
+    cluster: ClusterName,
+    backend: BackendId,
+    drone: DroneId,
+) {
+    let message = UpdateBackendStateMessage {
+        state,
+        cluster,
+        backend,
+        time: Utc::now(),
+        drone,
+    };
+    let nc = nc.clone();
+
+    while let Err(error) = nc.request(&message).await {
+        tracing::error!(?error, "Failed to update backend state, retrying.");
+        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
```

Comment:
If the controller doesn't `msg.respond(&())`, does NATS make `nc.request()` return an `Err`? I'm trying to understand how the controller's ack figures into this logic - and how often the drone resends the message if no ack is received.

Reply:
Yes, if the controller doesn't hear back within NATS' [request timeout](https://docs.rs/async-nats/latest/async_nats/struct.ConnectOptions.html#method.request_timeout) (default 10s), it will return a timeout error.

But more likely than timing out, it will error right away if it can't send the message, and then leave a 1s gap.